File: frontend/src/types/insights.ts
export type AudienceInsight = {
  id: number;
  summary: string;
  what_people_loved: string[];
  constructive_feedback: string[];
  recurring_complaints: string[];
  common_questions: string[];
  content_suggestions: string[];
  provider: string;
  model: string;
  source_comment_count: number;
  generated_at: string;
};

File: backend/src/veya/api/schemas/insights.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict


class AudienceInsightResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    summary: str
    what_people_loved: list[str]
    constructive_feedback: list[str]
    recurring_complaints: list[str]
    common_questions: list[str]
    content_suggestions: list[str]
    provider: str
    model: str
    source_comment_count: int
    generated_at: datetime
